File: frontend/app/src/main/java/com/iesvirgendelcarmen/secondlife/model/Product.kt
package com.iesvirgendelcarmen.secondlife.model

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize

@Parcelize
data class Product (
    var _id: String,
    var publisher: String,
    var title :String,
    var description :String,
    var price :Float,
    val images :MutableList<String>,
    val category: Category,
    val isSold :Boolean = false) :Parcelable

File: backend/docker/server/api/model/product.js
const mongoose = require('mongoose')

const Schema = mongoose.Schema

const productSchema = new Schema({
    publisher : {
        type: mongoose.ObjectId,
        required: true
    },
    category : {
        type: String,
        enum: ['MOTOR', 'INMOBILIARIA', 'EMPLEO', 'SERVICIOS', 'OTROS'],
        required: true,
        default: 'OTROS'
    },
    title : {
        type: String,
        required: true,
        trim: true
    },
    description : {
        type: String,
        required: true,
        trim: true
    },
    price : {
        type: Number,
        required: true
    },
    images : {
        type: [String]
    },
    isSold : {
        type: Boolean,
        default: false
    }
}, {versionKey : false})

module.exports = mongoose.model('Product', productSchema)